pkg/gardenlet/apis/config/validation/validation.go

```diff
@@ -24,44 +24,34 @@ import (
 )
 
 // ValidateGardenletConfiguration validates a GardenletConfiguration object.
-func ValidateGardenletConfiguration(cfg *config.GardenletConfiguration) field.ErrorList {
+func ValidateGardenletConfiguration(cfg *config.GardenletConfiguration, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	if cfg.Controllers != nil {
 		if cfg.Controllers.Shoot != nil {
-			allErrs = append(allErrs, ValidateShootControllerConfiguration(cfg.Controllers.Shoot, field.NewPath("controllers", "shoot"))...)
+			allErrs = append(allErrs, ValidateShootControllerConfiguration(cfg.Controllers.Shoot, fieldPath(fldPath, "controllers", "shoot"))...)
```

Comment:
Optional: you can directly use `Child` if you want. It will result in the same behavior as `fieldPath()` if `fldPath == nil`.
```suggestion
			allErrs = append(allErrs, ValidateShootControllerConfiguration(cfg.Controllers.Shoot, fieldPath.Child("controllers", "shoot"))...)
```

Reply:
I can't call `fieldPath.Child` (`fieldPath` is a function), I guess you mean `fldPath.Child`. I thought it would result in a nil pointer panic if `fldPath == nil` so I introduced the `fieldPath` function. However you are right - `Child` never uses its receiver to access fields, so there is no panic, and it actually behaves as if `NewPath` was called. 

I am still hesitant though - I find it unnatural and perhaps dangerous to call methods on nil receivers and depend on the implementation to not get a panic. Not sure if this behavior can be considered an implementation detail or not, the documentation also doesn't say much. Perhaps it's better to keep the function as a precaution?

Reply:
Finally I removed the function in favor of `fldPath.Child`, should be ok.